src/_pytest/python.py

```diff
@@ -1109,7 +1118,7 @@ def _idval_from_argname(argname: str, idx: int) -> str:
 
 
 @final
-@attr.s(frozen=True, slots=True, auto_attribs=True)
+@dataclasses.dataclass(frozen=True)
```

Comment:
note, we might wan to walk our frozen classes and replace type annotations for mutable things with type annotations for immutable variants to catch missuses of mutable field of "immutable" objects

Reply:
Do you mean annotating all fields `Final` instead of using `frozen`? I'm rather partial to `frozen` as it's much less noisy, so I prefer to keep it for now.